draft-ietf-gnap-core-protocol.md

```diff
@@ -2688,33 +2688,45 @@ This is often part of facilitating [interaction](#user-interaction), but it coul
 also be used to allow the AS and RC to continue negotiating the parameters of
 the [original grant request](#request). 
 
-To enable this ongoing negotiation, the AS returns a `continue` field 
+To enable this ongoing negotiation, the AS provides a continuation API to the RC.
+The AS returns a `continue` field 
 [in the response](#response-continue) that contains information the RC needs to
-continue this process with another request, including a URI to access
-as well as an optional access token to use during the continued requests.
-
-When the RC makes any calls to the continuation URL, the RC MUST present
-proof of the most recent key associated with this ongoing request
-by signing the request as described in {{binding-keys}}. The key in use will
-be either the key from [the initial request](#request-key) or its most recent
-rotation. 
+access this API, including a URI to access
+as well as an access token to use during the continued requests. 
+
+The access token is initially bound to the same key and method the RC used to make 
+the initial request. As a consequence,
+when the RC makes any calls to the continuation URL, the RC MUST present
+the access token as described in {{use-access-token}} and present
+proof of the RC's key (or its most recent rotation)
+by signing the request as described in {{binding-keys}}.
+
 \[\[ [See issue #85](https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/85) \]\]
 
-For example, here the RC makes a POST request and signs with detached
-JWS:
+For example, here the RC makes a POST request to a unique URI and signs 
+the request with detached JWS:
 
 ~~~
-POST /continue/80UPRY5NM33OMUKMKSKU HTTP/1.1
+POST /continue/KSKUOMUKM HTTP/1.1
+Authorization: GNAP 80UPRY5NM33OMUKMKSKU
 Host: server.example.com
 Detached-JWS: ejy0...
 ~~~
 
-If the AS includes an `access_token` in the `continue`
-response in {{response-continue}}, the RC MUST include the access token the
-request as described in {{use-access-token}}. Note that the access token
-is always bound to the RC's presented key (or its most recent rotation).
+The AS MUST be able to tell from the RC's request which specific ongoing request
+is being accessed. Common methods for facilitating this include using a unique, unguessable URL
```

Comment:
This is security critical (hence a MUST on the first line of the paragraph) and yet we are leaving this decision to the implementer. Why not pick one of these "common methods" and mandate it? Reducing the optionality is a **good thing** when it makes for more secure implementations.

Reply:
This is a question of API design and deployment. There is no optionality in the protocol here: it always includes a URL and an access token. We're allowing the AS to decide if it uses the access token alone, the URL, or a combination of the two in order to identify the ongoing request that the continuation request represents. If we do mandate something be the deciding factor, it should be the access token, but I'm not convinced that's really our business to say.

Reply:
Agreed, this is optionality at the implementation level, not the protocol. My push is to reduce the number of decisions implementers have to make, each of which could easily become a security issue.

Reply:
Agreed. We might include a note for implementers if we think some guidance would help (just providing examples of deployment and their impact to make it concrete).

Reply:
> This is security critical (hence a MUST on the first line of the paragraph) and yet we are leaving this decision to the implementer. Why not pick one of these "common methods" and mandate it? Reducing the optionality is a **good thing** when it makes for more secure implementations.

I agree with @yaronf's comments here on the optionality in implementation. This is NOT an access token in how it is usually used -- it is managing state for the AS to determine the request. A unique URL is the common mechanism. I also don't think the URL needs to be unique for each "continuation" -- it only needs to be unique for that grant request.